foundry/auto_self_nominate.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+use crate::config::load_config;
+use ccore::stake::Action::SelfNominate;
+use ccore::stake::{Banned, Candidates, Jail, CUSTOM_ACTION_HANDLER_ID};
+use ccore::{BlockId, ConsensusClient, Encodable, EngineSigner, SignedTransaction, UnverifiedTransaction};
+use ckey::PlatformAddress;
+use clap::ArgMatches;
+use codechain_types::transaction::{Action, Transaction};
+use primitives::Bytes;
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+const NEED_NOMINATION_UNDER_TERM_LEFT: u64 = 2;
+pub struct AutoSelfNomination {
+    client: Arc<dyn ConsensusClient>,
+    signer: EngineSigner,
```

Comment:
`EngineSigner` was created to use in Conseuss Engines. What do you think about creating a `Signer` struct that is similar to the `EngineSigner`?